Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -5068,4 +5068,40 @@ void Document::set_needs_to_refresh_scroll_state(bool b)
         paintable->set_needs_to_refresh_scroll_state(b);
 }
 
+Vector<JS::NonnullGCPtr<Range>> Document::find_matching_text(String const& query)
+{
+    if (!document_element() || !document_element()->layout_node())
+        return {};
+
+    Vector<JS::NonnullGCPtr<Range>> matches;
```

Comment:
This doesn't look safe. A `Vector<JS::NonnullGCPtr>` doesn't protect its elements from GC. Either use `MarkedVector` or a `Vector` of `JS::Handle`s :)